homeassistant/components/climate/zhong_hong.py

```diff
@@ -0,0 +1,229 @@
+"""
+Support for ZhongHong HVAC Controller.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.zhong_hong/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ATTR_OPERATION_MODE, PLATFORM_SCHEMA, STATE_COOL, STATE_DRY,
+    STATE_FAN_ONLY, STATE_HEAT, SUPPORT_FAN_MODE, SUPPORT_ON_OFF,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE, ClimateDevice)
+from homeassistant.const import (ATTR_TEMPERATURE, CONF_HOST, CONF_PORT,
+                                 EVENT_HOMEASSISTANT_STOP, TEMP_CELSIUS)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_GATEWAY_ADDRRESS = 'gateway_address'
+
+REQUIREMENTS = ['zhong_hong_hvac==1.0.9']
+EVENT_DEVICE_SETTED_UP = 'zhong_hong_device_setted_up'
+EVENT_ZHONG_HONG_HUB_START = 'zhong_hong_hub_start'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST):
+    cv.string,
+    vol.Optional(CONF_PORT, default=9999):
+    vol.Coerce(int),
+    vol.Optional(CONF_GATEWAY_ADDRRESS, default=1):
+    vol.Coerce(int),
+})
+
+
+async def async_setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the ZhongHong HVAC platform."""
+    from zhong_hong_hvac.hub import ZhongHongGateway
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    gw_addr = config.get(CONF_GATEWAY_ADDRRESS)
+    hub = ZhongHongGateway(host, port, gw_addr)
+    try:
+        devices = [
+            ZhongHongClimate(hub, addr_out, addr_in)
+            for (addr_out, addr_in) in hub.discovery_ac()
+        ]
+
+    except Exception as exc:
+        _LOGGER.error("ZhongHong controller is not ready", exc_info=exc)
+        raise PlatformNotReady
+
+    device_num = len(devices)
+    _LOGGER.debug("We got %s zhong_hong climate devices", device_num)
+    device_ready_num = 0
+    semaphore_lock = asyncio.Semaphore()
+
+    async def device_count(event):
+        """Count ready device and fire hub start event."""
+        nonlocal device_ready_num
+        with (await semaphore_lock):
+            device_ready_num += 1
+            _LOGGER.debug("%s zhong_hong device setted up", device_ready_num)
+            if device_ready_num >= device_num:
+                hass.bus.fire(EVENT_ZHONG_HONG_HUB_START)
+
+    async_remove_listener = hass.bus.async_listen(EVENT_DEVICE_SETTED_UP,
```

Comment:
Use our dispatch helper instead of the event bus. Here you connect a target string to a function. The function should be the one that checks if all devices have been added to home assistant.